frontend/src/components/Forms/MineRecordForm.js

```diff
@@ -65,39 +117,134 @@ export class MineRecordForm extends Component {
   removeField = (event, fields, index) => {
     event.preventDefault();
     fields.remove(index);
+    const removedMineType = this.props.mine_types[index].mine_tenure_type_code;
+    this.setState((prevState) => ({
+      usedTenureTypes: prevState.usedTenureTypes.filter(
+        (tenureTypes) => tenureTypes !== removedMineType
+      ),
+    }));
+  };
+
+  // addField allows users to create a max of 4 mine Types.
+  addField = (event, fields) => {
+    const totalTypes = this.props.currentMineTypes
```

Comment:
Probably not necessary, but I think this might be a bit clearer:
```javascript
const totalTypes = (this.props.currentMineTypes.length || 0) + fields.length;
```

It took me a few seconds to parse the ternary, but I think the above code is a bit more direct. Again, if you have a default value of `currentMineTypes` then it shouldn't be `undefined`

Reply:
Checking the lengths without the ternary or `|| 0` works great now that there is a default for `currentMineTypes`
``` const totalTypes = fields.length + this.props.currentMineTypes.length;```